src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -53,6 +53,26 @@ public RestoreCommand(RestoreRequest request)
             _request = request;
         }
 
+        public RestoreCommand(RestoreSummaryRequest summaryRequest)
```

Comment:
Making the lower level aware of the higher level wrapper here feels very tangled.

At minimum just pass the CollectorLogger and the request here instead of the summary. 

Ideally the higher level summary request would not exist at all and it would get the list of errors another way. Sharing these loggers seems wrong here, though duplicating the errors would take a lot more memory. The logic for getting errors and writing them to the assets file within RestoreCommand shouldn't depend on something from a higher level like this.